correctness/mockmetheus.py

```diff
@@ -0,0 +1,289 @@
+                for timeseries in result.get('timeseries', []):
+                    timeseries['labels'].sort(key=lambda label: label['name'])
+            return response
+        except Exception as e:
+            print(f"Error parsing response: {e}")
+            raise
+
+    def remote_write(self, record):
+        """
+        Sends a remote write request to the Prometheus connector.
+
+        Args:
+            record (list): The data to write, structured as a list of dictionaries.
+
+        Returns:
+            int: The HTTP status code of the write response.
+
+        Raises:
+            Exception: If there's an error in sending the request.
+        """
+        try:
+            # Construct WriteRequest from record
+            write_request = self.construct_write_request(record)
+            serialized_data = write_request.SerializeToString()
+            compressed_data = snappy.compress(serialized_data)
+            write_response = requests.post(
+                f"{self.CONNECTOR_URL}/write",
+                data=compressed_data,
+                headers=self.PROMETHEUS_HEADERS,
+                auth=self.BASIC_AUTH,
+            )
+            write_response.raise_for_status()
+            return write_response.status_code
+        except Exception as e:
+            print(f"Error sending remote write request: {e}")
+            raise
+
+    def parse_duration(self, duration_str):
+        """
+        Parses a Prometheus duration string and returns the duration in milliseconds.
+        """
+        pattern = r'(?:(\d+)h)?(?:(\d+)m)?(?:(\d+)s)?$'
+        match = re.match(pattern, duration_str)
+        if not match:
+            raise ValueError(f"Invalid duration format: {duration_str}")
+
+        hours, minutes, seconds = match.groups()
+        total_ms = 0
+        if hours:
+            total_ms += int(hours) * 60 * 60 * 1000
+        if minutes:
+            total_ms += int(minutes) * 60 * 1000
+        if seconds:
+            total_ms += int(seconds) * 1000
+        if total_ms == 0:
+            raise ValueError(f"Duration cannot be zero: {duration_str}")
+        return total_ms
+
+    def construct_read_request(self, query_string):
+        """
+        Creates a ReadRequest from a given PromQL query string, optionally with read hints.
+
+        Args:
+            query_string (str): The PromQL query string, optionally including read hints.
+
+        Returns:
+            ReadRequest: The constructed ReadRequest object.
+
+        Raises:
+            ValueError: If the query string format is invalid.
+        """
+        read_request = ReadRequest()
+
+        # Split the query into main part and hints part if '@' is present
+        parts = query_string.split('@', 1)
+        main_query = parts[0].strip()
+        hints_query = parts[1].strip() if len(parts) > 1 else None
+
+        metric_pattern = r'^\s*([a-zA-Z_:][a-zA-Z0-9_:]*)'            # Metric name
```

Comment:
These regex strings don't support [Prometheus functions](https://prometheus.io/docs/prometheus/latest/querying/functions/). For example, `sum(http_requests_total{method="GET"} @ 1609746000)`. This also means that for now we can't recreate the correctness "tests" that we [used to use](https://github.com/awslabs/amazon-timestream-connector-prometheus/blob/main/correctness/data/functions.txt). That may be fine for now if we're okay with not testing functions.